pkg/cli/api_address_builder.go

```diff
@@ -0,0 +1,52 @@
+package cli
+
+import (
+	"context"
+	"github.com/gorilla/websocket"
+	"net"
+	"net/http"
+	"net/url"
+)
+
+func (api *ApiClient) buildHttpClientAndAddress() (*http.Client, *url.URL, error) {
```

Comment:
Nitpick: Name this `buildHTTPClientAndURL `? Since, technically, this returns an URL, and not an address?

```suggestion
func (api *ApiClient) buildHTTPClientAndURL() (*http.Client, *url.URL, error) {
```